gno.land/cmd/gnoland/secrets_verify.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/privval"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/p2p"
+)
+
+type secretsVerifyCfg struct {
+	commonAllCfg
+}
+
+// newSecretsVerifyCmd creates the secrets verify command
+func newSecretsVerifyCmd(io commands.IO) *commands.Command {
+	cfg := &secretsVerifyCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "verify",
+			ShortUsage: "secrets verify [flags] [<key>]",
+			ShortHelp:  "verifies all Gno secrets in a common directory",
+			LongHelp: fmt.Sprintf(
+				"verifies the validator private key, the node p2p key and the validator's last sign state. Available keys: %s",
+				getAvailableSecretsKeys(),
+			),
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execSecretsVerify(cfg, args, io)
+		},
+	)
+}
+
+func (c *secretsVerifyCfg) RegisterFlags(fs *flag.FlagSet) {
+	c.commonAllCfg.RegisterFlags(fs)
+}
+
+func execSecretsVerify(cfg *secretsVerifyCfg, args []string, io commands.IO) error {
+	// Make sure the directory is there
+	if cfg.dataDir == "" || !isValidDirectory(cfg.dataDir) {
+		return errInvalidDataDir
+	}
+
+	// Verify the secrets key
+	if err := verifySecretsKey(args); err != nil {
+		return err
+	}
+
+	var key string
+
+	if len(args) > 0 {
+		key = args[0]
+	}
+
+	// Construct the paths
+	var (
+		validatorKeyPath   = filepath.Join(cfg.dataDir, defaultValidatorKeyName)
+		validatorStatePath = filepath.Join(cfg.dataDir, defaultValidatorStateName)
+		nodeKeyPath        = filepath.Join(cfg.dataDir, defaultNodeKeyName)
+	)
+
+	switch key {
+	case validatorPrivateKeyKey:
+		// Validate the validator's private key
+		_, err := readAndVerifyValidatorKey(validatorKeyPath, io)
+
+		return err
+	case validatorStateKey:
+		// Validate the validator's last sign state
+		validatorState, err := readAndVerifyValidatorState(validatorStatePath, io)
+		if err != nil {
+			return err
+		}
+
+		// Attempt to read the validator key
+		if validatorKey, err := readAndVerifyValidatorKey(validatorKeyPath, io); validatorKey != nil && err == nil {
+			// Validate the signature bytes
+			return validateValidatorStateSignature(validatorState, validatorKey.PubKey)
```

Comment:
imo, if this step is skipped, this should be logged with a warning

Reply:
Added warning:

[888712c](https://github.com/gnolang/gno/pull/1593/commits/888712c87179a2ed46c761590e610269fd812145)